akka-stream/src/main/scala/akka/stream/impl/io/SslTlsCipherActor.scala

```diff
@@ -415,10 +419,30 @@ private[akka] class SslTlsCipherActor(settings: ActorMaterializerSettings, sslCo
     }
   }
 
+  val hostnameVerifier = {
+    val sslConfig = SSLConfigFactory.parse(context.system.settings.config)
+    val systemImpl = context.system.asInstanceOf[ActorSystemImpl]
+    val v = systemImpl.dynamicAccess.createInstanceFor[HostnameVerifier](sslConfig.hostnameVerifierClass, Nil)
+      .getOrElse(throw new Exception("Unable to obtain hostname verifier for class: " + sslConfig.hostnameVerifierClass))
+
+    log.debug("hostnameVerifier: " + v)
+    v
+  }
   private def handshakeFinished(): Unit = {
     if (tracing) log.debug("handshake finished")
-    currentSession = engine.getSession
-    corkUser = false
+    val session = engine.getSession
+
+    //    println("session.getPeerCertificates = " + session.getPeerCertificates.toList)
+    println("session.getId = " + session.getId.toList)
+    println("session.getPeerHost = " + session.getPeerHost)
+    val hostname = hostInfo.map(_._1).getOrElse("WAT")
```

Comment:
not sure here: if we didn't get a hostname here... we should skip verification? // cc @wsargent @jrudolph